vowpalwabbit/v_array.h

```diff
@@ -172,40 +172,14 @@ struct v_array<T, typename std::enable_if<std::is_trivially_copyable<T>::value>:
     (--_end)->~T();
   }
 
-  VW_DEPRECATED("v_array::pop() is deprecated. Use pop_back(). This will be removed in VW 9.0.")
-  T pop()
-  {
-    T ret = back();
-    pop_back();
-    return ret;
-  }
-
-  VW_DEPRECATED("v_array::last() is deprecated. Use back(). This will be removed in VW 9.0.")
-  T last() const { return *(_end - 1); }
-
   bool empty() const { return _begin == _end; }
 
-  VW_DEPRECATED("v_array::decr() is deprecated. Use pop_back(). This will be removed in VW 9.0.")
-  void decr() { _end--; }
-
-  VW_DEPRECATED("v_array::incr() is deprecated. Use push_back(). This will be removed in VW 9.0.")
-  void incr()
-  {
-    if (_end == _end_array) reserve_nocheck(2 * capacity() + 3);
-    _end++;
-  }
   T& operator[](size_t i) const { return _begin[i]; }
   inline size_t size() const { return _end - _begin; }
   inline size_t capacity() const { return _end_array - _begin; }
 
-  // maintain the original (deprecated) interface for compatibility. To be removed in VW 10
-  VW_DEPRECATED(
-      "v_array::resize() is deprecated. Use reserve() instead. For standard resize behavior, use "
-      "resize_but_with_stl_behavior(). This will be removed in VW 9.0.")
-  void resize(size_t length) { reserve_nocheck(length); }
-
   // change the number of elements in the vector
-  // to be renamed to resize() in VW 10
```

Comment:
we're just changing the version we intended to do this on?

Reply:
As can be seen in the above removal the deprecation attribute and comment were out of sync. The deprecation attribute take precedence. Given that, this is updated to be in like with the rest of the changes in this file.